firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistributionLifecycleNotifier.java

```diff
@@ -20,14 +20,19 @@
 import androidx.annotation.GuardedBy;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
 import com.google.android.gms.tasks.Task;
 import com.google.android.gms.tasks.TaskCompletionSource;
-import com.google.android.gms.tasks.Tasks;
 import java.util.ArrayDeque;
 import java.util.Queue;
 
 class FirebaseAppDistributionLifecycleNotifier implements Application.ActivityLifecycleCallbacks {
 
+  /** A functional interface for a function that takes an activity and does something with it. */
+  interface ActivityConsumer {
+    void consume(Activity activity) throws FirebaseAppDistributionException;
```

Comment:
Looks good! Would like to chat about this consume stuff since I am not familiar with it.